app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/ConversionHostsSettings.js

```diff
@@ -8,22 +8,68 @@ import ConversionHostWizard from './components/ConversionHostWizard';
 import { FETCH_V2V_PROVIDERS_URL } from '../../../../../../redux/common/providers/providersConstants';
 
 class ConversionHostsSettings extends React.Component {
+  pollingInterval = null;
+  state = { hasMadeInitialFetch: false };
+
   componentDidMount() {
-    const { fetchProvidersAction, fetchProvidersUrl, fetchConversionHostsAction, fetchConversionHostsUrl } = this.props;
+    const { fetchProvidersAction, fetchProvidersUrl } = this.props;
     fetchProvidersAction(fetchProvidersUrl);
-    fetchConversionHostsAction(fetchConversionHostsUrl);
+    this.startPolling();
+  }
+
+  componentWillUnmount() {
+    this.stopPolling();
   }
 
+  componentDidUpdate(prevProps) {
+    // When a modal closes, reset the polling interval to see results immediately
+    if (this.pollingInterval && this.hasSomeModalOpen(prevProps) && !this.hasSomeModalOpen()) {
+      this.startPolling();
+    }
+  }
+
+  hasSomeModalOpen = (props = this.props) =>
+    props.conversionHostWizardMounted || props.conversionHostDeleteModalVisible;
+
+  startPolling = () => {
```

Comment:
Adding yet another place with a copy of this code?

`app/javascript/react/screens/App/Overview/Overview.js` , `app/javascript/react/screens/App/Mappings/Mappings.js` and `app/javascript/react/screens/App/Plan/Plan.js` all do the same thing here.... sounds like something that should be abstracted away

Reply:
@himdel I noticed that as well, but I was hesitant to refactor those as part of this PR since they are... well.. a bit fragile. I figured we should go back and revisit the polling abstraction later on, since we need to get this in for the build on March 16th. If you feel strongly about it though, I can take a look and see what it would take to abstract this away.

One reason I didn't do it right away is that this new polling implementation has some subtle differences, most notably the fact that it restarts automatically when a modal closes (instead of those modals forcing another check when they close of the same resources we poll for). So the refactor would involve changing that behavior too.

Reply:
Agreed @mturley :)

I don't think there's any pressing need to fix this in *this* PR, as long as things are moving forward.

Just wanted to make sure the pattern won't be spreading unchecked, better catch this sooner than later :)

Reply:
I agree. Opened https://github.com/ManageIQ/manageiq-v2v/issues/904 to track it, I'll put it on my list :)